tools/extras/check_dependencies.sh

```diff
@@ -87,28 +87,40 @@ if ! which awk >&/dev/null; then
   add_packages gawk gawk gawk
 fi
 
-if which python >&/dev/null ; then
+pythonok=true
+if ! which python2.7 >&/dev/null; then
+  echo "$0: python2.7 is not installed"
+  add_packages python2.7
+  pythonok=false
+fi
+
+if ! which python3 >&/dev/null; then
+  echo "$0: python3 is not installed"
+  add_packages python3
+  pythonok=false
+fi
+
+( 
+#Use a subshell so that sourcing env.sh does not have an influence on the rest of the script
+[ -f ./env.sh ] && . ./env.sh
+if $pythonok && ! which python2 >&/dev/null; then
+  mkdir -p $PWD/python
+  echo "$0: python2.7 is installed, but the python2 binary does not exist. Adding this to tools/env.sh"
+  ln -s $(which python2.7) $PWD/python/python2
+  echo "export PATH=$PWD/python:\${PATH}" >> env.sh
```

Comment:
you are creating python2 link, but we usually use shebang python
it would be OK I guess in case when you would be working on PR that changes the shebangs to python2 -- without that I'm not sure it would fix any issue

Reply:
Yes, my thought has been that we switch to use python2 shebangs, either at once or gradually. I just did not want to include it in this same PR, but that would be coming.